include/smack/BoogieAst.h

```diff
@@ -83,6 +84,13 @@ class BoolLit : public Expr {
   void print(std::ostream& os) const;
 };
 
+class RModeLit : public Expr {
```

Comment:
Where is this used?

Reply:
Don't you have to update all floating-point operations that are generated by smack to use `$rmode`?

Reply:
This is used by the change to SmackRep.cpp above. I update the floating-point operations in [this pull request]<https://github.com/smackers/smack/pull/328>.

Reply:
I see. Ok, this should then probably have a check to make sure that only valid rounding modes are used. I guess we should define them as string constants in `Naming.h` as well. And then I think that `val` should not be a string type, but rather an `enum` type.